src/Views/BucketPanel/BucketDetail.test.tsx

```diff
@@ -111,4 +112,46 @@ describe("BucketDetail", () => {
 
     expect(removeButton).toBeUndefined();
   });
+
+  it("should display rename icon for entries", async () => {
+    const detail = mount(
+      <BucketDetail
+        client={client}
+        permissions={{ fullAccess: false, write: ["BucketWithData"] }}
+      />,
+    );
+    const renameIcon = await waitUntilFind(detail, { title: "Rename entry" });
+
+    expect(renameIcon.hostNodes().length).toEqual(2);
+  });
+
+  it("should hide rename icon for entries if no permissions", async () => {
+    const detail = mount(
+      <BucketDetail client={client} permissions={{ fullAccess: false }} />,
+    );
+    const renameIcon = await waitUntilFind(detail, { title: "Rename entry" });
+
+    expect(renameIcon).toBeUndefined();
+  });
+
+  it("should open rename modal on rename icon click", async () => {
+    const detail = mount(
+      <MemoryRouter>
+        <BucketDetail client={client} permissions={{ fullAccess: true }} />
+      </MemoryRouter>,
+    );
+    await waitUntil(() => detail.update().find(".ant-table-row").length > 0);
+
+    const rows = detail.find(".ant-table-row");
+    const emptyEntryRow = rows.at(1);
+
+    const renameIcon = emptyEntryRow.find('span[title="Rename entry"]');
+    renameIcon.simulate("click");
+
+    const renameModal = detail.find('div[data-testid="rename-modal"]');
+    const removeModal = detail.find('div[data-testid="delete-modal"]');
+
+    expect(renameModal.exists()).toBe(true);
+    expect(removeModal.exists()).toBe(false);
```

Comment:
let's check if the renameEntry method was called.

Reply:
I was struggling to simulate the input update with antd this morning but finally found a solution ! Now it should be good.